### PR TITLE
Add range set/install functionality

### PR DESCRIPTION
**Pull request by AndrewBezold**
### What was wrong?

Solc version could not be set or installed from a range of versions
### How was it fixed?

The highest compatible solc version is set based on a string with syntax matching that of the pragma version line in a solidity contract.  If none exist, the highest compatible version is installed and set.
#### Cute animal picture

alt-cute-animal-picture

### CODE REVIEW

**Comment by AndrewBezold** [comment]
Don't merge this yet, I think I can make this better.  Found the supported pragma version syntax, I should be able to make it support all of that.

**Comment by AndrewBezold** [comment]
Ok, ready for review.  I made the _pragma functions separate only because I didn't want to lose backwards compatibility, and it would've taken some bit of effort to try to distinguish between the two strings with incompatible syntax.

**Comment by iamdefinitelyahuman** [comment]
My schedule today is pretty full-on but I'll review asap

File: solcx/install.py [hunk 1]
```diff
@@ -79,6 +112,58 @@ def install_solc(version=None):
...
+    raise ValueError("Compatible solc version does not exist")
+
+def _compare_versions(v1, v2, operator='='):
+    v1 = v1.lstrip('v')
+    v2 = v2.lstrip('v')
+    v1_split = [int(i) for i in v1.split('.')]
+    v2_split = [int(i) for i in v2.split('.')]
+    if operator == '' or operator == None:
+        operator = '='
+    if operator == '=':
```
Comment (iamdefinitelyahuman) [inline solcx/install.py]: Using the [operator](https://docs.python.org/3.7/library/operator.html) module you could shorten this block of if statements and make it a lot more readable.

Comment (AndrewBezold) [reply solcx/install.py]: I'm not sure how that would shorten this.  It feels like it'd be a one-for-one swap exactly the same length.

**Comment by iamdefinitelyahuman** [comment]
Awesome, thanks for this!

**Pull request merged**